### PR TITLE
Add test for negotiated DataChannel.OnOpen

### PR DESCRIPTION
**Pull request by edaniels**
5dc7245 was missing a test for this

### CODE REVIEW

File: datachannel_test.go [hunk 1]
```diff
@@ -155,6 +155,63 @@ func TestDataChannel_Open(t *testing.T) {
...
+			Negotiated: &negotiated,
+			Ordered:    &ordered,
+		})
+		assert.NoError(t, err)
+
+		answerDC.OnMessage(func(msg DataChannelMessage) {
+			go func() {
+				// Wait a little bit to ensure all messages are processed.
+				time.Sleep(100 * time.Millisecond)
+				done <- true
```
Comment (ashellunts) [inline datachannel_test.go:193]: Looks like this is not used, right?

File: datachannel_test.go [hunk 2]
```diff
@@ -155,6 +155,63 @@ func TestDataChannel_Open(t *testing.T) {
...
+		report := test.CheckRoutines(t)
+		defer report()
+
+		offerPC, answerPC, err := newPair()
+		if err != nil {
+			t.Fatalf("Failed to create a PC pair for testing")
+		}
+
+		done := make(chan bool)
+		answerOpenCalls := make(chan bool, 2)
```
Comment (ashellunts) [inline datachannel_test.go:169]: Why 2? Please use named constant.

File: datachannel_test.go [hunk 3]
```diff
@@ -155,6 +155,63 @@ func TestDataChannel_Open(t *testing.T) {
...
+				done <- true
+			}()
+		})
+		answerDC.OnOpen(func() {
+			answerOpenCalls <- true
+		})
+
+		offerDC.OnOpen(func() {
+			offerOpenCalls <- true
+			e := offerDC.SendText("Ping")
```
Comment (ashellunts) [inline datachannel_test.go:202]: Is this used in a test?

File: datachannel_test.go (hunk 2 above)
Comment (edaniels) [reply datachannel_test.go:169]: the existing test does the same thing for the sake of ensuring that an open happens only once but gives it the capacity for two in order to check for that failure.

File: datachannel_test.go (hunk 1 above)
Comment (edaniels) [reply datachannel_test.go:193]: done is used in `closePair` in both tests.

File: datachannel_test.go (hunk 3 above)
Comment (edaniels) [reply datachannel_test.go:202]: Yes, `answerDC` uses this message in order to send to `done`

File: datachannel_test.go (hunk 2 above)
Comment (edaniels) [reply datachannel_test.go]: added constants

File: datachannel_test.go [hunk 4]
```diff
@@ -155,6 +157,63 @@ func TestDataChannel_Open(t *testing.T) {
...
+			ID:         &dataChannelID,
+			Negotiated: &negotiated,
+			Ordered:    &ordered,
+		})
+		assert.NoError(t, err)
+
+		answerDC.OnMessage(func(msg DataChannelMessage) {
+			go func() {
+				// Wait a little bit to ensure all messages are processed.
+				time.Sleep(100 * time.Millisecond)
```
Comment (ashellunts) [inline datachannel_test.go:194]: Sorry, I don't know exactly this code, but curious what exactly do you wait here?
Is it possible to write the test without sleep?

Comment (edaniels) [reply datachannel_test.go:194]: I’m copying this from the test above so I’m not sure. I’d rather keep this the same and then later change both test later.

**Review by ashellunts — APPROVED** [review]

**Comment by Sean-Der** [comment]
Thanks for reviewing @ashellunts and sorry I got pulled away @edaniels :(

**Pull request merged**